src/main/kotlin/nl/connectplay/scoreplay/controllers/UserController.kt

```diff
@@ -195,4 +196,32 @@ class UserController(
             call.respond(HttpStatusCode.InternalServerError)
         }
     }
+
+    suspend fun handleUpdateUserAsync(call: ApplicationCall) { // the call: Application is a small package that holeds the request and respons
+        val userId = call.parameters["id"]?.toIntOrNull()
+            ?: return call.respond(HttpStatusCode.BadRequest, "User ID is not a number")// read the id and cover it to int if possible if null badrequest
+        val updateDto = call.receiveNullable<UserUpdateDto>()
+            ?: return call.respond(HttpStatusCode.BadRequest, "Invalid or no update date") // receiveNullable checks if the UserUpdateDto is valid
+
+        try {
+            userRepository.updateUserAsync(userId, updateDto) // send the update to the UserRepository
+            // if there is no user found trow NotFound message 404
+            // null is not allowed for JSON respond
+            val updatedUser = userRepository.getUserByIdAsync(userId)
+                ?: return call.respond(HttpStatusCode.NotFound, "No user found after update")
+            call.respond(HttpStatusCode.OK, updatedUser) // send a HTTP Ok response back to the client with the updated user data
+        } catch (e: SQLException) {
+            // check if the error message is about UNIQUE or duplicate values in the database
```

Comment:
nee. catch hier een [SQLIntegrityConstraintViolationException](https://docs.oracle.com/javase/8/docs/api/java/sql/SQLIntegrityConstraintViolationException.html)

Een error message in een exception kan op elk moment veranderen

Je zou in een catch-block daaronder nog de algemene SQLException op kunnen vangen.